### PR TITLE
Secrets Service

### PR DESCRIPTION
**Pull request by ben-toogood**

### CODE REVIEW

**Comment by ben-toogood** [comment]
Just making a few minor changes

File: secrets/service/handler/secrets.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+// Handler implements the secrets service interface
+type Handler struct {
+	name   string
+	secret string
+	store  store.Store
+}
+
+// pathJoiner is the character used to join the path when writing to the store
+var pathJoiner = "/"
```
Comment (domwong) [inline secrets/service/handler/secrets.go:43]: style: Might as well make this a `const` rather than var

File: secrets/service/handler/secrets.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	iv := text[:aes.BlockSize]
+	text = text[aes.BlockSize:]
+	cfb := cipher.NewCFBDecrypter(block, iv)
+	cfb.XORKeyStream(text, text)
+	data, err := base64.StdEncoding.DecodeString(string(text))
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
```
Comment (domwong) [inline secrets/service/handler/secrets.go:172]: Tests? Worth checking that decrypt and encrypt do the right thing

**Comment by ben-toogood** [comment]
> LGTM, only minor comments. I'm assuming we don't enforce any namespacing of secrets at this level e.g. right now it's possible for anyone to overwrite a secret, no namespaces or anything like that.

Thanks Dom. I'll add some tests for encrypt / decrypt and fix that const/var typo. Right now this is just going to be for us to use, so I didn't implement namespacing / multi-tenancy. I suspect we'll add this once we offer secrets as a paid service.

**Pull request merged**